examples/medplum-demo-bots/src/deduplication/find-matching-patients.ts

```diff
@@ -0,0 +1,118 @@
+import { BotEvent, MedplumClient, createReference, getReferenceString, resolveId } from '@medplum/core';
+import { ListEntry, Patient, RiskAssessment, Task } from '@medplum/fhirtypes';
+
+/**
+ * This Bot listens for changes to a `Patient` resource and searches for potential patient duplicates.
+ * The matching algorithm used is to search for exact matches:
+ *  - first name
+ *  - last name
+ *  - date of birth
+ *  - postal code
+ *
+ * See: https://www.medplum.com/docs/fhir-datastore/patient-deduplication#matching-rules
+ * for more potential matching rules
+ *
+ * @param medplum - The Medplum client instance.
+ * @param event - The BotEvent containing the Patient resource.
+ */
+export async function handler(medplum: MedplumClient, event: BotEvent<Patient>): Promise<void> {
+  //This bot should only be triggered by a Patient resource Subscription only
+  const srcPatient = event.input;
```

Comment:
What's the hypothetical risk of race conditions here?  If two merge operations get started for the same Patient in rapid succession, how should users ensure that data integrity is maintained?

Reply:
The hypothetical risk is high. Managing race conditions over record merges is a known hard problem, and even the `$merge` doesn't address it. 

In practice, this should be handled at the workflow level. There are a few different ways that this could be handled: 
- Modify the merge bot to operate on _all_ `RiskAssessments` assigned to a given target patient, rather than on each `RiskAssessment` individually. This would keep the merges sequential within the Bot. 
- Make sure all matches for a given target patient are reviewed by the same person, and have the reviewer-facing UI force the reviewer to handle all merges for a given target in sequence

Reply:
Added a comment re: race conditions at the beginning of teh doc